src/views/Search.vue

```diff
@@ -89,6 +91,30 @@ export default defineComponent({
     })
   },
   methods: {
+    async scanProduct(){
+      const modal = await modalController
+        .create({
+          component: Scanner,
+        });
+        modal.onDidDismiss()
+        .then((result) => {
+          //result : value of the scanned barcode/QRcode
+          const viewSize =  process.env.VUE_APP_VIEW_SIZE;
+          const viewIndex =  0;
+          const payload = {
+            viewSize,
+            viewIndex,
+            queryString: '*' + result.role + '*'
+          }
+          if (result) {
+            this.store.dispatch("product/findProduct", payload);
```

Comment:
I think instead of writing all this logic again, we can call `getProducts()` method from here, WDYT?

Reply:
Updated the method getProducts and call the getProducts method instead of writing the logic again.